app/main.py

```diff
@@ -1,3 +1,24 @@
 class OnlineCourse:
-    # write your code here
-    pass
+    course_dict = {}
+
+    def __init__(self, name, description, weeks):
+        self.name = name
+        self.description = description
+        self.weeks = weeks
+        self.course_dict["name"] = name
+        self.course_dict["description"] = description
+        self.course_dict["weeks"] = weeks
+
+    def days_to_weeks(days):
+        if days % 7 == 0:
+            return days // 7
+        else:
+            return (days + 7) // 7
+
+    @classmethod
+    def from_dict(cls, course_dict):
+        return OnlineCourse(
```

Comment:
you should write `cls` instead of `OnlineCourse` here. If you inherited this class, your method would not work correctly in child classes